DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfiguration/RegistrationPromptsController.cs

```diff
@@ -103,88 +108,137 @@ public IActionResult AddRegistrationPromptSelectPrompt(AddRegistrationPromptSele
-            ModelState.AddModelError
-            (
-                nameof(EditRegistrationPromptViewModel.Answer),
-                $"The complete list of answers must be less than 4000 characters. The new answer can be maximum {remainingLength} characters long."
-            );
+            if (saveToTempData)
+            {
+                UpdateTempDataWithAnswersModelValues(model);
+            }
 
-            ModelState.Remove(nameof(EditRegistrationPromptViewModel.Options));
-            model.Options = NewlineSeparatedStringListHelper.SplitNewlineSeparatedList(model.OptionsString);
+            return View(model);
         }
 
-        private IActionResult EditRegistrationPromptPostRemovePrompt(EditRegistrationPromptViewModel model, int index)
+        private IActionResult RegistrationPromptAnswersPostRemovePrompt
+            (RegistrationPromptAnswersViewModel model, int index, bool saveToTempData = false)
         {
             IgnoreAddNewAnswerValidation();
 
-            var (optionsString, options) =
+            var optionsString =
                 NewlineSeparatedStringListHelper.RemoveStringFromNewlineSeparatedList(model.OptionsString!, index);
 
-            SetModelOptions(model, optionsString, options);
+            SetRegistrationPromptAnswersViewModelOptions(model, optionsString);
+
+            if (saveToTempData)
+            {
+                UpdateTempDataWithAnswersModelValues(model);
+            }
+
             return View(model);
         }
 
-        private IActionResult EditRegistrationPromptPostSave(EditRegistrationPromptViewModel model)
+        private IActionResult AddRegistrationPromptConfigureAnswersPostNext(RegistrationPromptAnswersViewModel model)
         {
             IgnoreAddNewAnswerValidation();
 
             if (!ModelState.IsValid)
             {
-                model.Options = NewlineSeparatedStringListHelper.SplitNewlineSeparatedList(model.OptionsString);
                 return View(model);
             }
 
-            customPromptsService.UpdateCustomPromptForCentre(User.GetCentreId(), model.PromptNumber, model.Mandatory, model.OptionsString);
+            UpdateTempDataWithAnswersModelValues(model);
 
+            // TODO: HEEDLS-454 - redirect to next page
             return RedirectToAction("Index");
         }
 
+        private void SetRegistrationPromptAnswersViewModelOptions
+            (RegistrationPromptAnswersViewModel model, string optionsString)
+        {
+            ModelState.Remove(nameof(RegistrationPromptAnswersViewModel.OptionsString));
+            model.OptionsString = optionsString;
+
+            ModelState.Remove(nameof(RegistrationPromptAnswersViewModel.Answer));
+            model.Answer = null;
+        }
+
+        private void AddTotalStringLengthRegistrationPromptAnswersViewModelError
+            (RegistrationPromptAnswersViewModel model)
+        {
+            var remainingLength = 4000 - (model.OptionsString?.Length - 2 ?? 0);

```

Comment:
Is it possible for the options string to have a length of less than 2?

Reply:
Technically yes, but not at the point where this would be a problem. This code will only be reached if (optionsString + '\r\n' + newAnswerString) > 4000. Which given the newAnswerString has a max length of 100 isn't possible until the optionsString has length greater than 3900

Reply:
You could add a if optionsString < 4000 return statement to the top of this method to make this extra safe :)

Reply:
I don't think that is the condition we want. Have added a check to return if the options string is null or has length less than 2. model.OptionsString < 4000 is always true. and optionsString (the updated one) it is always false at this point.